### DISCUSSION TITLE
Add a way to change the reputation o gains or loose a player

### DISCUSSION THREAD

**Discussion by Gamer73proprro** [SmartlyDressedGames/Unturned-3.x-Community#5357] [Ideas]
Sorry for my bad English or for using this to communicate, but is there a native way to track and change the reputation a player gains or loses? I tried using events like "UnturnedPlayerEvents.OnPlayerDeath" and creating my own reputation system, but the event thinks about doubling experience like: if you kill another player you lose -25 reputation, but when I hit the event it doubles to -50 or changes to +25, and if the other player is a villain or a model, the values ​​change a lot, another example, the normal thing would be to kill another player to get -25 but if it is a villain you get -85, and with the paragon you get +75, the values ​​vary constantly so I could not guarantee that they are the precise values ​​but it is an estimate.
I have also noticed that on some occasions without trying to manipulate the reputation it doubles anyway. I've had this problem with both the original version of RocketMod and LDM (Legally-Distinct-Missile). Maybe it's a code problem, but even when I run other plugins that have nothing to do with it, this error usually happens. I would greatly appreciate some kind of help or guidance, or definitely the direct addition of a native way to change this aspect.
Apologies in advance if I published this in the wrong place or the problem lacks foundations, please know that I am just starting out on this and a helping hand would be very useful to me. Have a nice day, afternoon or evening wherever you are!

**Comment by DanielWillett** [comment]
It sounds like you are subscribing to the event but never unsubscribing from it then reloading the plugin. this can cause events to run multiple times.

Usually you should do stuff like this:
```cs
class Plugin
{
 override Load()
 {
    UnturnedPlayerEvents.OnPlayerDeath += OnDeath;
 } 
 override Unload()
 {
    UnturnedPlayerEvents.OnPlayerDeath -= OnDeath;
 } 
}
```

Also note anonymous methods like below can not be unsubscribed from so you should always try to use named methods.
```cs
UnturnedPlayerEvents.OnPlayerDeath +=
player =>
{
   Logger.Log(player + " died");
}
```

**Comment by SDGNelson** [comment]
Thanks for your kindly worded post Gamer73proprro, congratulations on your progress so far! And thank you for your helpful answer DanielWillett - that does sound quite plausible.

**Comment by Gamer73proprro** [comment]
heyah! i think thats not the error cause i suscribe and unsuscribe in my plugin on the event like this you say?:
protected override void Load()
{
    Instance = this;
    Rocket.Core.Logging.Logger.LogWarning("this is the load message!");
    UnturnedPlayerEvents.OnPlayerDeath += OnPlayerDied;

}

protected override void Unload()
{
    UnturnedPlayerEvents.OnPlayerDeath -= OnPlayerDied;
    Rocket.Core.Logging.Logger.Log("this is the unload message!");
}
